autoprecompiles/src/execution_profile.rs

```diff
@@ -111,6 +159,7 @@ impl PgoCollector {
     }
 
     fn increment(&self, pc: u64) {
+        self.pc_list.write().unwrap().push(pc);
         self.pc_index_map[(pc - self.pc_base) as usize / self.step as usize]
             .fetch_add(1, Ordering::Relaxed);
```

Comment:
everything only depends on the pc list, so for simplicity I'd say we should only collect the list, then construct `ExecutionProfile::from_pcs(pcs)` from it.